armory/data/adversarial/resisc_adversarial_patch.py

```diff
@@ -0,0 +1,82 @@
+"""
+TensorFlow Dataset for resisc45 attacked by Adversarial Patch with adv/clean splits
+"""
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+
+_DESCRIPTION = """\
+NWPU-RESISC45 adversarial image dataset for DenseNet121
+DenseNet121 training on benign images: 
+    architecture name, batch_size, initial learning rate, optimizer configuration, loss, metrics
+
+AdversarialPatch
+    max_iter 1000
+    scale_min, scale_max 0.4
+    learning_rate 2.0
+    (By definition targeted)
+"""
+
+_URL = "https://armory-public-data.s3.us-east-2.amazonaws.com/imagenet-adv/ILSVRC12_ResNet50_PGD_adversarial_dataset_v1.0.tfrecords"  # to be updated
+_PATH = "armory/data/data/adversarial/RESISC45_DenseNet121_AdvPatch_adversarial_dataset_v0.1.tfrecords"
+
+
+class ImagenetAdversarial(tfds.core.GeneratorBasedBuilder):
+    """ RESISC45_DenseNet121_AdvPatch_adversarial_dataset_v0.1.tfrecords """
+
+    VERSION = tfds.core.Version("1.0.0")
+
+    def _info(self):
+        return tfds.core.DatasetInfo(
+            builder=self,
+            description=_DESCRIPTION,
+            features=tfds.features.FeaturesDict(
+                {
+                    "image": tfds.features.Tensor(shape=[224, 224, 3], dtype=tf.uint8),
```

Comment:
The dimensions here don't match the input dimensions of the original dataset, so if a user tries to use their own model that doesn't do the same preprocessing you have for densenet121, this will fail.

Reply:
For this scenario, resizing the adversarial examples up and then down again to attack the DenseNet probably would only minimally impact white-box attack efficacy, due to having 1:1 aspect ratio.

For scenarios, like video, where the aspect ratio of the original data (240:320) is different from the delivered adversarial example (112:112), we may need to create two different tfds scripts, one that keeps the adversarial examples at (112:112) and one that resizes to (240:320). The reason is that MARS preprocessing takes a center crop before resizing down.